### ISSUE TITLE
add support for WAMP (Web Application Messaging Protocol)

### ISSUE THREAD

**Issue by CircleCode** [hoaproject/Websocket#26]
[WAMP](http://wamp.ws/) is getting more and more attention.
We should add support for WAMP into Hoa\WebSocket.

**Comment by CircleCode** [comment]
once supported, Hoa should be added to http://wamp.ws/implementations/

**Comment by Hywan** [comment]
Any volunteers? /cc @hoaproject/hoackers

**Comment by jignesh87** [comment]
Hello..Hywan
I wanna streaming video is it posiible?.
Thanks.

**Comment by Hywan** [comment]
@jignesh87 Absolutely, it is possible, but it is not the goal of `Hoa\Websocket` for several reasons. Firstly, the WebSocket protocol is based on TCP and this kind of network is not suitable for video streaming since it is a “connected” mode. You are rather likely to use UDP, which is a “not-connected” mode. Secondly, WebSocket is designed for full-duplex connections, which means that the server can talk to the client and *vice versa*. In the case of a video streaming, you don't want the client to send data to the server. Finally, I encourage you to [come on IRC](http://hoa-project.net/Community.html#IRC) to ask your question. We are here to help you.

**Comment by jignesh87** [comment]
Thank you Hywan for reply. I will come on IRC soon.
Thank you again.:)

**Comment by Neofox** [comment]
Hello, any news on the implementation of WAMP?
Thanks

**Comment by Hywan** [comment]
@Neofox We have zero demand for that yet, and it is a huge work. We are not going to implement this specification if no one asks it :-).
